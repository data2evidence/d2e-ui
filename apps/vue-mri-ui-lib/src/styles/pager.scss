.pager-container {
  text-align: center;
  $enabledY: 0px;
  $disabledY: -60px;
  button {
    font-size: 12px;
    min-width: 20px;
    height: 20px;
    border-style: none;
    background-color: transparent;
    cursor: pointer;
  }
  button.pager-nav {
    background-position-y: $enabledY;
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAHgAAABQCAAAAAD38O5VAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAACdFJOUwAAdpPNOAAAAAlwSFlzAAAOwwAADsMBx2+oZAAAABl0RVh0U29mdHdhcmUAUGFpbnQuTkVUIHYzLjUuODc7gF0AAADmSURBVGje7ZZBDsMgDAR5La/Nfzi6h7pC8ppWhlS2op2DDyuHiSwIaY2Qh9L7rCa1IQaLFnc1m2qnDTuGGKx63OU88bY3ZEbxgTdiLiROG3Xe5ko5ToQQQuozxqwTkU+B9HviPug6xmitDROKiBZIf0Tuk0MlIN72hswoPvBGzIXEaaPO21wpx4kQQkh93h90vGTuvCRch3bKP69F8V5n2Xjnj4CGVnzgjZgLidNGnbe5Uo4TIYSQ+lzXrCa1IQaLFnc1m2qnDS8MMVj1uMt54m1vyIziA2/EXEicNuq8zZVynAh5EC9PRpS/Kg5kUgAAAABJRU5ErkJggg==');
  }
  button.pager-disabled {
    background-position-y: $disabledY;
  }
  button.pager-ll {
    background-position-x: 0px;
  }
  button.pager-l {
    background-position-x: -40px;
  }
  button.pager-r {
    background-position-x: -60px;
  }
  button.pager-rr {
    background-position-x: -100px;
  }
  button.pager-nav:not(.pager-disabled):hover {
    background-color: $MriButtonHover;
    background-position-y: -20px;
  }
  button.current-page-button {
    background-color: $MriDropDownListItemSelected;
  }
  button.page-button:hover {
    color: $MriLightestText;
    background-color: $MriButtonHover;
  }
}
