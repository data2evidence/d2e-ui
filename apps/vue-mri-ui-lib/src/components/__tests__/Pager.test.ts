import { shallowMount } from '@vue/test-utils'
import { createStore } from 'vuex'
import pager from '../Pager.vue'

describe('Pager.vue', () => {
  let store

  beforeEach(() => {
    store = createStore({
      getters: {
        getText: (modulestate, modulegetters) => (key, param) => {
          return key
        },
      },
    })
  })

  it('sets the correct default data', () => {
    const wrapper = shallowMount(pager as any)
    const defaultData = wrapper.props()

    expect(defaultData).toMatchSnapshot()
  })
  it('renders a div with class pager-container if there is at least 2 pages', () => {
    const wrapper = shallowMount(pager as any, {
      global: {
        plugins: [store],
      },
      props: { rowCount: 40 },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('not rendered when there is only a single page', () => {
    const wrapper = shallowMount(pager as any, {
      global: {
        plugins: [store],
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('emits goPage with page number', () => {
    const wrapper = shallowMount(pager as any, {
      global: {
        plugins: [store],
      },
    })
    wrapper.vm.$emit('goPage', 10)
    // assert event has been emitted
    expect(wrapper.emitted().goPage).toBeTruthy()

    // assert event count
    expect(wrapper.emitted().goPage[0]).toEqual([10])
  })
})
