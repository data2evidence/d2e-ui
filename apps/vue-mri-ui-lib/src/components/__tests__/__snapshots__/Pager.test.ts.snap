// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager.vue not rendered when there is only a single page 1`] = `
<div class="pager-disabled" style="padding-top: 20px;">
  <!--v-if-->
</div>
`;

exports[`Pager.vue renders a div with class pager-container if there is at least 2 pages 1`] = `
<div class="" style="padding-top: 20px;">
  <div class="pager-container"><button title="MRI_PA_PAGER_FIRST_PAGE" class="pager-nav pager-ll pager-disabled"><span>&nbsp;</span></button><button title="MRI_PA_PAGER_PREV_PAGE" class="pager-nav pager-l pager-disabled"><span>&nbsp;</span></button><button class="page-button current-page-button"><span>1</span></button><button class="page-button"><span>2</span></button><button title="MRI_PA_PAGER_NEXT_PAGE" class="pager-nav pager-r"><span>&nbsp;</span></button><button title="MRI_PA_PAGER_LAST_PAGE" class="pager-nav pager-rr"><span>&nbsp;</span></button></div>
</div>
`;

exports[`Pager.vue sets the correct default data 1`] = `
Object {
  "currentPage": 1,
  "pageSize": 20,
  "rowCount": 1,
  "v2": false,
}
`;
